features/storage/TESTS/blockdevice/general_block_device/main.cpp

```diff
@@ -209,6 +213,140 @@ void test_multi_threads()
     TEST_ASSERT_EQUAL(0, err);
 }
 
+void test_get_erase_value()
+{
+    utest_printf("\nTest BlockDevice::get_erase_value()..\n");
+
+    BlockDevice *block_device = BlockDevice::get_default_instance();
+    TEST_ASSERT_NOT_NULL(block_device);
+
+    int err = block_device->init();
+    TEST_ASSERT_EQUAL(0, err);
+
+    // Check erase value
+    int erase_value = block_device->get_erase_value();
+    if (erase_value < 0) {
+        utest_printf("\nblock_device->get_erase_value() returned %d, which means the erase value is unknown\n", erase_value);
+        goto exit;
+    }
+
+    // Assuming that get_erase_value() returns byte value
+    //  "If get_erase_value returns a non-negative byte value"
+    TEST_ASSERT(erase_value < 255);
+
+    // Allocate buffer for read test data
+    bd_addr_t start_address = (rand() * block_device->get_erase_size()) % (block_device->size());
+    bd_size_t read_buf_size = block_device->get_erase_size();
+    uint8_t *read_buf = (uint8_t*)malloc(read_buf_size);
+    TEST_ASSERT_NOT_NULL(read_buf);
+
+    // Erase given memory region
+    utest_printf("\nerasing erased memory region");
+    err = block_device->erase(start_address, read_buf_size);
```

Comment:
I think you better first program random data and only then erase it.
Otherwise if the erased area was already "erased" even before the test started you the test will pass successfully even if currently erase fails..

Reply:
Good idea, I'll do that.